packages/block-library/src/column/edit.js

```diff
@@ -77,10 +78,18 @@ function ColumnEdit( {
 export default compose(
 	withSelect( ( select, ownProps ) => {
 		const { clientId } = ownProps;
-		const { getBlockOrder } = select( 'core/block-editor' );
+		const {
+			getBlockOrder,
+			getBlockRootClientId,
+			getTemplateLock,
+		} = select( 'core/block-editor' );
+
+		// Retrieve the parent of the Columns Block.
+		const rootParentClientId = getBlockRootClientId( getBlockRootClientId( clientId ) );
 
 		return {
 			hasChildBlocks: getBlockOrder( clientId ).length > 0,
+			parentLock: getTemplateLock( rootParentClientId ),
```

Comment:
This is not the lock of the parent it is the lock of the grand-parent/lock of columns parent. Should we name the prop columnsParentLock?